### ISSUE TITLE
Works with S31 but can it report energy?

### ISSUE THREAD

**Issue by Scott-St** [sandysound/orvibo-b25-server#13]
I can confirm this is working with the newer S31 plug.  The plug also has power monitoring.  I am looking for a way that it would also report the energy use back in the json object.

Going to keep digging around but its hard to follow the code I didnt make so I didnt know if it was easy to add in.

This is the packet that the energy is in.  It is a different cmd number then the state (42)

`Socket ->  {"cmd":127,"serial":215,"uid":"<UID removed>","timestamp":1549919006,"energy":"0.00"}`

If there is anything I can do to help, or where I can go to add this let me know.

**Comment by sandysound** [comment]
Hey Scott, 
Sorry I just saw your message. Good to know that it works with the new S31 plug. I don't have any of them to test with but I could give you some pointers.
You could try adding  this to the OrviboPacket.js
```
pkt.prototype.getEnergy = function() {
    return this.payloadJSON.energy;
};
```
Then in the heartbeatHandler in Orvibo.js you could try adding this to the packet data like:
```
let pkData = Object.assign({}, socketData, {
        serial: plugPacket.getSerial(),
        uid: plugPacket.getUid()
       energy: plugPacket.getEnergy()
    });
```
Then it should appear in the socket data :)
let me know how you go and if you get it working.

**Comment by Scott-St** [comment]
I tried to put this code in but I think there is more to it on the side of the switch because I only get one power measurement when I connect then I dont see anymore.  Also, I only see 0.0 so I am not sure its even right.  I'm thinking its a parameter that needs to be queried.  I'll have to investigate it more when I have time but for now its working well to switch it on and off after I modified the function to be able to send either a 1 or 0 to turn it on or off instead of just  toggle.

Appreciate the code you have written and its confirmed that the S31 is working with the basic on and off function.
